stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanFlow.kt

```diff
@@ -0,0 +1,110 @@
+package com.stripe.android.stripecardscan.cardscan
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Rect
+import androidx.lifecycle.LifecycleOwner
+import com.stripe.android.camera.CameraPreviewImage
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopAggregator
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopState
+import com.stripe.android.camera.framework.AggregateResultListener
+import com.stripe.android.camera.framework.AnalyzerLoopErrorListener
+import com.stripe.android.camera.framework.AnalyzerPool
+import com.stripe.android.camera.framework.ProcessBoundAnalyzerLoop
+import com.stripe.android.stripecardscan.payment.ml.SSDOcr
+import com.stripe.android.stripecardscan.payment.ml.SSDOcrModelManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+internal data class SavedFrame(
+    val hasOcr: Boolean,
+    val frame: SSDOcr.Input,
+)
+
+internal data class SavedFrameType(
+    val hasOcr: Boolean,
+)
+
+internal abstract class CardScanFlow(
+    private val scanErrorListener: AnalyzerLoopErrorListener,
+) : AggregateResultListener<MainLoopAggregator.InterimResult, MainLoopAggregator.FinalResult> {
```

Comment:
Sorry I missed this earlier. Can we please extend `ScanFlow` here for when we move CardImageVerifyActivity to an XML definition? It would be nice to be able to share this.